pkg/datarepair/queue/queue.go

```diff
@@ -13,9 +15,9 @@ import (
 
 // RepairQueue is the interface for the data repair queue
 type RepairQueue interface {
-	Enqueue(qi *pb.InjuredSegment) error
-	Dequeue() (pb.InjuredSegment, error)
-	Peekqueue(limit int) ([]pb.InjuredSegment, error)
+	Enqueue(ctx context.Context, qi *pb.InjuredSegment) error
```

Comment:
Is using context for all this methods is necessary?

Reply:
Yes, usually should be passing the `ctx` around.